src/Libraries/CoreNodeModels/DropDown.cs

```diff
@@ -203,6 +202,8 @@ protected override bool UpdateValueCore(UpdateValueParams updateValueParams)
                     Warning(Dynamo.Properties.Resources.NothingIsSelectedWarning);
                 }
 
+                OnNodeModified();
```

Comment:
Is this function called when `SelectedIndex ` property is set to a different value?

Reply:
Sorry, I didn't get it then, how does moving `OnNodeModified` here work in that case?

Reply:
When exactly is UpdateValueCore called in this case?

Reply:
Ok, I see, by moving it out of `SelectedIndex`, you are avoiding chances of an infinite loop, since `SelectedIndex` is changed in `PopulateItems`, which is incidentally called here: https://github.com/DynamoDS/DynamoRevit/blob/a1c5384651133ebab113dd0abfef5bd256ef51f0/src/Libraries/RevitNodesUI/GenericClasses.cs#L194 inside the `BuildOutputAst` function, am I right?
I think I would agree that calling `PopulateItems` in `BuildOutputAst` is a bad idea in the first place.

Reply:
Correct. 
Since it's been that way for a while (7 years!), perhaps we could fix it back to working as it did before #14122 and look at improving in a separate task?